streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -118,14 +120,53 @@
      * query the value of the key on a parallel running instance of your Kafka Streams application.
      * <p>
      *
+     * @param predicate     a filter {@link Predicate} that is applied to each record
+     * @param materialized  a {@link Materialized} that describes how the {@link StateStore} for this {@code KTable}
```

Comment:
nit: `for this` -> `for the resulted`. Ditto elsewhere.

Reply:
Also for each pair of the overloaded functions of, e.g., `filter(predicate, materialized)` and `filter(predicate)`. We should probably state in the javadocs of the latter that, the default state store materialization mechanism will be used and abstracted away from the user, and also the backing state store of the KTable will not be queryable.